modules/nextflow/src/main/groovy/nextflow/container/CharliecloudBuilder.groovy

```diff
@@ -60,8 +68,37 @@ class CharliecloudBuilder extends ContainerBuilder<CharliecloudBuilder> {
     @Override
     CharliecloudBuilder build(StringBuilder result) {
         assert image
+        def imageStorage = Paths.get(image).parent.parent
+        def imageToRun = String
+
+        if (!writeFake) {
+            // define image to run, if --write-fake is not used this is a copy of the image in the current workDir
+            imageToRun = '"$NXF_TASK_WORKDIR"/container_' + image.split('/')[-1]
```

Comment:
How the `NXF_TASK_WORKDIR` variable is expected to be resolved?

Reply:
This would end up in `.command.sh` and should be expanded fine, unless I am missing something. The idea here is to create a temporary container inside the task work directory, which is exclusive to the task.

Reply:
Got it